wsd/FileServer.cpp

```diff
@@ -1751,7 +1751,20 @@ FileServerRequestHandler::ResourceAccessDetails FileServerRequestHandler::prepro
     Poco::replaceInPlace(preprocess, std::string("%ENABLE_WELCOME_MSG%"), enableWelcomeMessage);
     Poco::replaceInPlace(preprocess, std::string("%AUTO_SHOW_WELCOME%"), autoShowWelcome);
 
+    std::string userAccessibility;
+    Poco::JSON::Object::Ptr jsonBrowserSetting = new Poco::JSON::Object();
+    const std::string browserSetting = form.get("browser_setting", "");
+    if (!browserSetting.empty()) {
+        if (JsonUtil::parseJSON(browserSetting, jsonBrowserSetting)) {
+            JsonUtil::findJSONValue<std::string>(jsonBrowserSetting, "accessibility", userAccessibility);
+        }
+    }
+
     std::string enableAccessibility = stringifyBoolFromConfig(config, "accessibility.enable", false);
+    if (!userAccessibility.empty()) {
+        enableAccessibility = userAccessibility;
```

Comment:
We should make sure `userAccessibility` can only be true|false, here it will inject any value it receives without any check.

Reply:
Yes, I believe the parseJSON does the prerequisite validation. Anyway, I will check it. Thanks.